browser-test/src/checkbox.test.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  AdminPrograms,
+  AdminQuestions,
+  ApplicantQuestions,
+  loginAsAdmin,
+  loginAsGuest,
+  logout,
+  selectApplicantLanguage,
+  startSession,
+  resetSession,
+} from './support'
+
+describe('Checkbox question for applicant flow', () => {
+  let pageObject
+
+  beforeAll(async () => {
+    const {page} = await startSession()
+    pageObject = page
+  })
+
+  afterEach(async () => {
+    await resetSession(pageObject)
+  })
+
+  describe('single checkbox question', () => {
+    let applicantQuestions
+    const programName = 'test program for single checkbox'
+
+    beforeAll(async () => {
+      // As admin, create program with single checkbox question.
+      await loginAsAdmin(pageObject)
+      const adminQuestions = new AdminQuestions(pageObject)
+      const adminPrograms = new AdminPrograms(pageObject)
+      applicantQuestions = new ApplicantQuestions(pageObject)
+
+      await adminQuestions.addCheckboxQuestion({
+        questionName: 'checkbox-color-q',
+        options: ['red', 'green', 'orange', 'blue'],
+        minNumChoices: 1,
+        maxNumChoices: 2,
+      })
+      await adminPrograms.addAndPublishProgramWithQuestions(
+        ['checkbox-color-q'],
+        programName,
+      )
+
+      await logout(pageObject)
+    })
+
+    it('with single checked box submits successfully', async () => {
+      await loginAsGuest(pageObject)
+      await selectApplicantLanguage(pageObject, 'English')
+
+      await applicantQuestions.applyProgram(programName)
+      await applicantQuestions.answerCheckboxQuestion(['blue'])
+      await applicantQuestions.clickNext()
+
+      await applicantQuestions.submitFromReviewPage(programName)
+    })
+
+    it('with no checked boxes does not submit', async () => {
+      await loginAsGuest(pageObject)
+      await selectApplicantLanguage(pageObject, 'English')
+
+      await applicantQuestions.applyProgram(programName)
+
+      // No validation errors on first page load.
+      const checkBoxError = '.cf-applicant-question-errors'
+      expect(await pageObject.isHidden(checkBoxError)).toEqual(true)
+
+      // Click next without selecting anything.
+      await applicantQuestions.clickNext()
+
+      // Check checkbox error and required error are present.
+      expect(await pageObject.isHidden(checkBoxError)).toEqual(false)
+      const checkboxId = '.cf-question-checkbox'
+      expect(await pageObject.innerText(checkboxId)).toContain(
```

Comment:
This is a bit brittle, because if the text changed, we'd have to update it here too. Alternatively, we could add an id to the required error and check that instead. It felt a bit weird to add an id just for testing though.

Reply:
I think this is fine. The developer will easily catch it when they run the tests before merging.

Ideally we could reference a shared value but because it spans languages we can't.

Reply:
Ok sounds good, thanks for your input!